metrics/cloudwatch2/cloudwatch2.go

```diff
@@ -0,0 +1,239 @@
+// Package cloudwatch2 emits all data as a StatisticsSet (rather than
+// a singular Value) to CloudWatch via the aws-sdk-go-v2 SDK.
+package cloudwatch2
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/cloudwatchiface"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/convert"
+	"github.com/go-kit/kit/metrics/internal/lv"
+)
+
+const (
+	maxConcurrentRequests = 20
+)
+
+// CloudWatch receives metrics observations and forwards them to CloudWatch.
+// Create a CloudWatch object, use it to create metrics, and pass those metrics as
+// dependencies to the components that will use them.
+//
+// To regularly report metrics to CloudWatch, use the WriteLoop helper method.
+type CloudWatch struct {
+	mtx                   sync.RWMutex
+	sem                   chan struct{}
+	namespace             string
+	svc                   cloudwatchiface.CloudWatchAPI
+	counters              *lv.Space
+	logger                log.Logger
+	numConcurrentRequests int
+}
+
+type option func(*CloudWatch)
+
+// WithLogger sets the Logger that will recieve error messages generated
+// during the WriteLoop
+func WithLogger(logger log.Logger) option {
```

Comment:
I find it's nice if functional options describe the default behavior if the option _isn't_ invoked. In this case, it might be nice to add: "By default, no logger is used."

Reply:
Fixing...